File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.py
def solution(n):
    end = int(n**0.5)
    nums = [i for i in range(n+1)]
    
    for i in range(2, n+1):
        for i in range(2*i, n+1, i):
            nums[i] = 0
    
    return len([i for i in nums if i != 0][1:])

File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.swift
import Foundation

func solution(_ n:Int) -> Int {
    let end = Int(sqrt(Double(n))) + 1
    var isPrime = Array(repeating: true, count: n + 1)
    (isPrime[0], isPrime[1]) = (false, false)
    
    for i in 2...end {
        for j in stride(from: i + i, to: n + 1, by: i) {
            isPrime[j] = false
        }
    }

    return isPrime.filter { $0 == true }.count
    
}